src/charm.py

```diff
@@ -124,28 +126,65 @@ def __init__(self, *args):
                 return False
         if not self._database_is_available():
-            self.unit.status = WaitingStatus("Waiting for the database to be available")
             return False
         if not self._get_database_uri():
-            self.unit.status = WaitingStatus("Waiting for database URI")
             return False
         if not self._container.exists(path=BASE_CONFIG_PATH) or not self._container.exists(
             path=CERTS_DIR_PATH
         ):
-            self.unit.status = WaitingStatus("Waiting for storage to be attached")
             return False
         if not _get_pod_ip():
-            self.unit.status = WaitingStatus("Waiting for pod IP address to be available")
             return False
         return True
 
+    def _on_collect_unit_status(self, event: CollectStatusEvent):  # noqa C901
+        """Check the unit status and set to Unit when CollectStatusEvent is fired.
+
+        Args:
+            event: CollectStatusEvent
+        """
+        if not self.unit.is_leader():
+            # NOTE: In cases where leader status is lost before the charm is
+            # finished processing all teardown events, this prevents teardown
+            # event code from running. Luckily, for this charm, none of the
+            # teardown code is necessary to perform if we're removing the
+            # charm.
+            event.add_status(BlockedStatus("Scaling is not implemented for this charm"))
+            return
+        if not self._container.can_connect():
+            event.add_status(WaitingStatus("Waiting for container to be ready"))
+            return
+        for relation in [DATABASE_RELATION_NAME, "certificates"]:
+            if not self._relation_created(relation):
+                event.add_status(BlockedStatus(f"Waiting for {relation} relation to be created"))
+                return
+        if not self._database_is_available():
+            event.add_status(WaitingStatus("Waiting for the database to be available"))
+            return
+        if not self._get_database_uri():
+            event.add_status(WaitingStatus("Waiting for database URI"))
+            return
+        if not self._container.exists(path=BASE_CONFIG_PATH) or not self._container.exists(
+            path=CERTS_DIR_PATH
+        ):
+            event.add_status(WaitingStatus("Waiting for storage to be attached"))
+            return
+        if not _get_pod_ip():
+            event.add_status(WaitingStatus("Waiting for pod IP address to be available"))
+            return
+        if self._csr_is_stored() and not self._get_current_provider_certificate():
+            event.add_status(WaitingStatus("Waiting for certificates to be stored"))
+            return
+        if not self._nrf_service_is_running():
+            event.add_status(WaitingStatus("Waiting for NRF service to start"))
+            return
+        event.add_status(ActiveStatus())
+
```

Comment:
In line 194, could we log that charm is  not ready for configuration ?
```
if not self.ready_to_configure():
     return
```

Reply:
Done.